apis/offloading/v1alpha1/namespaceoffloading_types.go

```diff
@@ -0,0 +1,167 @@
+	SomeFailedOffloadingPhaseType OffloadingPhaseType = "SomeFailed"
+	// AllFailedOffloadingPhaseType -> there was an error during creation of all remote Namespaces.
+	AllFailedOffloadingPhaseType OffloadingPhaseType = "AllFailed"
+)
+
+// NamespaceMappingStrategyType represents different strategies to map local and remote namespace names.
+type NamespaceMappingStrategyType string
+
+const (
+	// EnforceSameNameMappingStrategyType -> the remote namespace is assigned the same name of the local one
+	// (the creation may fail in case of conflicts).
+	EnforceSameNameMappingStrategyType NamespaceMappingStrategyType = "EnforceSameName"
+	// DefaultNameMappingStrategyType -> the remote namespace is assigned a default name which ensures uniqueness
+	// and avoids conflicts (localNamespaceName-localClusterID).
+	DefaultNameMappingStrategyType NamespaceMappingStrategyType = "DefaultName"
+)
+
+// PodOffloadingStrategyType represents different strategies to offload pods in this Namespace.
+type PodOffloadingStrategyType string
+
+const (
+	// LocalPodOffloadingStrategyType -> the pods in this namespace can be scheduled on the local cluster only
+	// (i.e. no pod offloading occurs).
+	LocalPodOffloadingStrategyType PodOffloadingStrategyType = "Local"
+	// RemotePodOffloadingStrategyType -> the pods in this namespace can be scheduled on remote clusters only, possibly
+	// filtered through the ClusterSelector field.
+	RemotePodOffloadingStrategyType PodOffloadingStrategyType = "Remote"
+	// LocalAndRemotePodOffloadingStrategyType -> the pods in this namespace can be scheduled on both the local
+	// and remote clusters, the latter possibly filtered through the ClusterSelector field.
+	LocalAndRemotePodOffloadingStrategyType PodOffloadingStrategyType = "LocalAndRemote"
+)
+
+// RemoteNamespaceConditionType represents different conditions that a remote namespace could assume.
+type RemoteNamespaceConditionType string
+
+// These are valid conditions of a remote namespace.
+const (
+	// NamespaceOffloadingRequired, informs user if his namespace has been offloaded on this cluster or not.
+	NamespaceOffloadingRequired RemoteNamespaceConditionType = "OffloadingRequired"
+	// NamespaceReady, remote Namespace is correctly created and ready to be used.
+	NamespaceReady RemoteNamespaceConditionType = "Ready"
+)
+
+// RemoteNamespaceConditions list of RemoteNamespaceCondition.
+type RemoteNamespaceConditions []RemoteNamespaceCondition
+
+// RemoteNamespaceCondition contains details about state of remote namespace.
+type RemoteNamespaceCondition struct {
+	// Type of remote namespace controller condition.
+	Type RemoteNamespaceConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// +optional
+	Message string `json:"message,omitempty"`
```

Comment:
You should add a comment to those. Optional is not required in combination with omitempty